compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -389,17 +388,14 @@ class MapAllocatorCache {
         const uptr HeaderPos = AllocPos - HeadersSize;
         const uptr MaxAllowedFragmentedBytes =
             MaxAllowedFragmentedPages * PageSize;
-        if (HeaderPos > CommitBase + CommitSize)
+        if (HeaderPos > CommitBase + CommitSize || HeaderPos < CommitBase)
           continue;
```

Comment:
Let's only do the change of `MaxReleasedCachePages` in this pull request. The change of `AllocPos` to `roundDown(HeaderPos, PageSize)` still introduce 1 page difference. It's better to do it separately